docs/setup.rst

```diff
@@ -0,0 +1,106 @@
+Installation
+============
+
+The easiest way to install Fuel is using the Python package manager pip. Fuel
+isn't listed yet on the Python Package Index (PyPI), so you will have to grab it
+directly from GitHub.
+
+.. code-block:: bash
+
+   $ pip install git+git://github.com/mila-udem/fuel.git
+
+This will give you the cutting-edge development version. The latest stable
+release is in the ``stable`` branch and can be installed as follows.
+
+.. code-block:: bash
+
+   $ pip install git+git://github.com/mila-udem/fuel.git@stable
+
+If you don't have administrative rights, add the ``--user`` switch to the
+install commands to install the packages in your home folder. If you want to
+update Fuel, simply repeat the first command with the ``--upgrade`` switch
+added to pull the latest version from GitHub.
+
+.. warning::
+
+   Pip may try to install or update NumPy and SciPy if they are not present or
+   outdated. However, pip's versions might not be linked to an optimized BLAS
+   implementation. To prevent this from happening make sure you update NumPy
+   and SciPy using your system's package manager (e.g.  ``apt-get`` or
+   ``yum``), or use a Python distribution like Anaconda_, before installing
+   Fuel. You can also pass the ``--no-deps`` switch and install all the
+   requirements manually.
+
+   If the installation crashes with ``ImportError: No module named
+   numpy.distutils.core``, install NumPy and try again again.
+
+
+Requirements
+------------
+Fuel's requirements are
+
+* PyYAML_, to parse the configuration file
+* six_, to support both Python 2 and 3 with a single codebase
+* h5py_ and PyTables_ for the HDF5 storage back-end
+* pillow_, providing PIL for image preprocessing
+* Cython_, for fast extensions
+* pyzmq_, to send data across processes
+* picklable_itertools_, for picklable iterators
+* SciPy_, to read from MatLab's .mat format
+* requests_, to download canonical datasets
+
+.. _Anaconda: https://store.continuum.io/cshop/anaconda/
+.. _nose2: https://nose2.readthedocs.org/
```

Comment:
You don't mention this anywhere. Probably best to say it's optional "to run the tests".